packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,623 @@
+	}
+
+	/**
+	 * Calculates and draws the tickmarks with a CSS gradient style
+	 *
+	 * @private
+	 */
+	_drawDefaultTickmarks(step, max, min) {
+		if (!this.tickmarks || !this.step) {
+			return;
+		}
+
+		// Convert number values to strings to let the CSS do calculations better
+		// rounding/subpixel behavior" and the most precise tickmarks distribution
+		const maxStr = String(max);
+		const minStr = String(min);
+		const stepStr = String(step);
+		const tickmarkWidth = "1px";
+
+		// There is a CSS bug with the 'currentcolor' value of a CSS gradient that does not
+		// respect the variable for more than one theme. It has to be set here for now.
+		const currentTheme = getTheme();
+		const currentColor = this.TICKMARK_COLOR_MAP[currentTheme];
+
+		this._tickmarksAmount = `${maxStr - minStr} / ${stepStr}`;
+		this._hiddenTickmarks = false;
+
+		// Transparent CSS gradient background
+		const tickmarksGradientBase = `linear-gradient(to right, ${currentColor} ${tickmarkWidth}, transparent 0) `;
+
+		// Draw the tickmarks as a patern over the gradient background
+		const tickmarksGradientdPattern = `0 center / calc((100% - ${tickmarkWidth}) / (${this._tickmarksAmount})) 100% repeat-x`;
+
+		// Combine to get the complete CSS background gradient property value
+		this._tickmarksBackground = `${tickmarksGradientBase + tickmarksGradientdPattern}`;
+
+		// If labelsInterval is specified draw labels for the necessary tickmarks
+		if (this.labelInterval > 0) {
+			this._drawDefaultLabels();
+		}
+	}
+
+	/**
+	 * Calculates the labels amout, width and text and creates them
+	 *
+	 * @private
+	 */
+	_drawDefaultLabels() {
+		const labelInterval = this.labelInterval;
+		const step = this.step;
+		const newNumberOfLabels = (this.max - this.min) / (step * labelInterval);
+
+		// If the required labels are already rendered
+		if (newNumberOfLabels === this._oldNumberOfLabels) {
+			return;
+		}
+
+		this._oldNumberOfLabels = newNumberOfLabels;
+		this._labelWidth = 100 / newNumberOfLabels;
+		this._labelItems = [];
+
+		// If the step value is not a round number get its precision
+		const stepPrecision = SliderBase._getDecimalPrecisionOfNumber(step);
+
+		// numberOfLabels below can be float so that the "distance betweenlabels labels"
+		// calculation to be precize (exactly the same as the distance between the tickmarks).
+		// That's ok as the loop stop condition is set to an integer, so it will practically
+		// "floor" the number of labels anyway.
+		for (let i = 0; i <= newNumberOfLabels; i++) {
+			// Format the label numbers with the same decimal precision as the value of the step property
+			const labelItemNumber = ((i * step * labelInterval) + this.min).toFixed(stepPrecision);
+			this._labelItems.push(document.createTextNode(labelItemNumber));
+		}
+	}
+
+	/**
+	 * Normalizes a new <code>step</code> property value.
+	 * If tickmarks are enabled recreates them according to it.
+	 *
+	 * @private
+	 */
+	_setStep(step) {
+		if (step === 0) {
```

Comment:
For the "step" property you can create a data type, similar to the Integer type, under the `packages/main/src/types` folder

```js
import DataType from "./DataType.js";

class SliderStep extends DataType {
	static isValid(value) {
		return typeof value === "number" && value > 0;
	}
}

export default SliderStep;
```

```js
step: {
   type: SliderStep,
   defaultValue: 1,
}
```

The framework will use this isValid method when the component is initilized with the provided states and if the user sets an invalid value or value not in the required range, the framework would fallback to the default.

Reply:
As we have come to the conclusion that the value of 0 will be a valid value for the step property we've decided that it's type will remain integer